### PR TITLE
Harden process command allowlists

### LINKED ISSUE #832: Security (defense in depth): process command allowlist check is skipped when allowlist is empty
## Problem

`validate_process_command_allowlist_hir` (`stage1/crates/axiomc/src/hir.rs:11117-11153`) short-circuits to `Ok(())` when `capabilities.process_commands.is_empty()`:

```rust
if capabilities.process_commands.is_empty() {
    return Ok(());
}
```

So if any caller ever produces a `CapabilityConfig` with `process = true, process_commands = []`, the HIR check accepts arbitrary (including non-literal) command expressions silently.

Today the manifest parser (`manifest.rs:821`) makes that state hard to reach — it errors if `process` is `true` with an empty list, and `manifest.rs:424` warns. So this is a defense-in-depth issue, not an exploitable bypass yet. But the HIR-level invariant is the wrong shape: the validator should fail closed.

## Impact

Any future code path that constructs `CapabilityConfig` outside the manifest parser (test harnesses, programmatic embedding, the planned semantic-layer `capability` declarations in #779/#780) inherits the gap.

## Success criteria

- [ ] `validate_process_command_allowlist_hir` rejects non-literal command expressions even when `process_commands` is empty (or returns an internal error since the manifest layer should already have prevented this state).
- [ ] `CapabilityConfig` exposes the unrestricted vs. allowlisted state as an explicit enum (e.g., `Allowlist(Vec<String>)` vs. `Unrestricted`), so empty-list-means-unrestricted is unrepresentable.
- [ ] A unit test in `hir.rs` covers the `process = true, allowlist = []` shape and asserts the validator's chosen failure mode.

### PR DESCRIPTION
**Pull request by jmcte**
## Summary

- Model process command policy as an explicit `Unrestricted` vs. non-empty `Allowlist` enum instead of treating an empty command vector as unrestricted.
- Fail closed in both source and HIR process allowlist validators if an invalid empty allowlist shape is ever constructed outside manifest parsing.
- Add a HIR regression test for `process = true` with an empty allowlist and a dynamic `process_status` command expression.

## Governing Issue

Closes #832

## Validation

- [x] Relevant local checks passed
  - `cargo fmt -p axiomc --manifest-path stage1/Cargo.toml -- --check`
  - `cargo test -p axiomc process_allowlist_empty_shape_fails_closed_for_dynamic_command --manifest-path stage1/Cargo.toml`
  - `cargo test -p axiomc process --manifest-path stage1/Cargo.toml`
  - `git diff --check`
  - `AXIOM_PYTHON_EXIT_ISSUE_STATES_JSON='{"266":"closed","267":"closed","268":"closed","269":"closed","270":"closed","271":"closed"}' bash scripts/ci/run-fast-checks.sh`
- [x] Required PR checks are expected to satisfy `CI Gate`
- [x] Skipped checks are explained below

Additional probe:

- `cargo test -p axiomc capability --manifest-path stage1/Cargo.toml` was not used as a gate because it exposed an unrelated existing failure: `stage1_project_rejects_stdlib_crypto_sign_without_crypto_capability` reports missing `std/crypto_sign.ax`.

## Bootstrap Governance

- [x] Changes are scoped to the linked issue
- [x] Contributor or PR guidance changes are reflected in `CONTRIBUTING.md`, `.github/PULL_REQUEST_TEMPLATE.md`, and `docs/bootstrap/onboarding.md` when applicable
- [ ] Auto-merge is enabled, or GitHub plan-limit evidence is recorded and the fallback merge-readiness policy applies
- [x] No real secrets, runtime auth, or machine-local env files are committed

## Semantic Layer Checklist

- [x] Semantic node types added/changed are listed, or this PR does not touch semantic-layer behavior
- [x] Schemas added/changed are listed, or this PR does not touch schemas
- [x] Evidence added/changed is listed, or this PR does not touch evidence
- [x] Rust capture check is satisfied, or this PR is backend-only and marks backend-specific behavior
- [x] Agent-facing inspection impact is described, or there is no inspection impact

## Flow Contract

- Owner lane: security
- Repair owner: Codex
- Autonomy class: agent-executed
- Risk class: security-sensitive defense-in-depth

## Flow Merge Readiness

- [x] Every blocker has a next actor and next action
- [x] No active blocking requested changes remain
- [ ] Non-author approval is present when required
- [ ] Auto-merge is appropriate when gates pass

## Merge Automation

- [ ] Auto-merge is enabled, or the reason it is unavailable or unsafe is noted below

## Notes

- Auto-merge and approval state need to be evaluated after GitHub creates the required check runs.

### CODE REVIEW

**Review by athena-omt — CHANGES_REQUESTED** [review]
Requesting changes: the process allowlist hardening is good, but the unrestricted-process audit/SBOM path still loses the required rationale, so `axiomc caps --json` can surface an unsafe process grant without the justification that should accompany it. Please wire the process unsafe rationale through the audit output before merging. ([github.com](https://github.com/OMT-Global/axiom/pull/865/files))

**Review by athena-omt — APPROVED** [review]
Verified: the audit path now threads `unsafe_rationale` through `capability_sbom`/`caps --json`, so the unrestricted process grant no longer drops the justification. The new regression test covers the empty-allowlist + dynamic `process_status` case. No remaining code-level blockers.

File: stage1/crates/axiomc/src/manifest.rs [hunk 1]
```diff
@@ -378,11 +435,23 @@ pub fn capability_descriptors(config: &CapabilityConfig) -> Vec<CapabilityDescri
             configured_root: None,
             effective_root: None,
             package_root: None,
-            unsafe_unrestricted: *kind == CapabilityKind::Env && config.env_unrestricted,
+            unsafe_unrestricted: match *kind {
+                CapabilityKind::Env => config.env_unrestricted,
+                CapabilityKind::Process => {
+                    config.process && config.process_commands.is_unrestricted()
```
[the remark this answers is not in the document]
Comment (jmcte) [reply stage1/crates/axiomc/src/manifest.rs:441]: @codex please address

**Comment by jmcte** [comment]
Rebased this branch onto current `origin/main` so PR Fast CI uses the authenticated Python-exit issue-state check.

Validation:
- `cargo fmt -p axiomc --manifest-path stage1/Cargo.toml -- --check`
- `cargo test -p axiomc --lib process_command --manifest-path stage1/Cargo.toml`
- `cargo test -p axiomc --lib capability_sbom_preserves_unrestricted_process_unsafe_rationale --manifest-path stage1/Cargo.toml`
- `bash scripts/ci/test-pr-fast-ci-workflow.sh`
- `git diff --check`

**Review by pheidon — APPROVED** [review]
